src/Datadog.Trace/HttpOverStreams/HttpMessage.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+using System.Text;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.HttpOverStreams
+{
+    internal abstract class HttpMessage
+    {
+        private static readonly Vendors.Serilog.ILogger Log = DatadogLogging.For<HttpMessage>();
+        private static readonly UTF8Encoding Utf8Encoding = new UTF8Encoding(encoderShouldEmitUTF8Identifier: false);
+
+        public HttpMessage(HttpHeaders headers, IHttpContent content)
+        {
+            Headers = headers;
+            Content = content;
+        }
+
+        public HttpHeaders Headers { get; }
+
+        public IHttpContent Content { get; }
+
+        public int? ContentLength => int.TryParse(Headers.GetValue("Content-Length"), out int length) ? length : (int?)null;
+
+        public string ContentType => Headers.GetValue("Content-Type");
+
+        public Encoding GetContentEncoding()
```

Comment:
`ContentType` is accessed multiple times in this method accessing every time to the public HttpHeaders instance. It's better to introduce a local var for ContentType to ensure we are using always the same value and reducing the Get overhead.

Reply:
Fixed